design_docs/config-file_reorganization.md

```diff
@@ -0,0 +1,108 @@
+<h1> Config File Reorganization <br>
+Xylar Asay-Davis<br>
+date: 01-29-2017<br>
+</h1>
+<h2> Summary </h2>
+This document describes various efforts to clean up the structure of the MPAS-Analysis config file.  The idea is to create a template config file that will replace `config.analysis` as well as a number of example config files designed to make use of various MPAS and ACME runs on various machines.  The reorganization should make the analysis easier for users to modify and run.
+
+<h1> Requirements </h1>
+
+<h2> Requirement: a simple way of turning on and off individual analysis modules <br>
+Date last modified: 2017/01/29 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+There should be a simple, intuitive method for turning on and off individual analysis modules (e.g. `ocean/ohc_timeseries`).  This should replace the current approach of having a boolean `generate` flag for each analysis module in a separate config section.  Preferably, there should be an equivalent method for turning on and off analysis modules from the command line that overrides that in the config file.
+
+<h2> Requirement: there should be a simplified template for config files <br>
+Date last modified: 2017/01/29 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+The current example config file is specific to a run on Edison and should be made into a general template.  Simplifications should be made to the template so that it can more easily and intuitively be modified for several analyses.  Example config files should also be added for analyzing several existing runs on several different machines.
+
+<h2> Requirement: removal of ACME specific config options <br>
+Date last modified: 2017/01/29 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+To the extent possible, ACME-specific config options such as `casename` and `ref_casename_v0` should be generalized in a way that is also appropriate for MPAS-only runs.
+
+<h2> Requirement: optional reference dates for each model run or set of observations<br>
+Date last modified: 2017/01/29 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+Currently, there is an option `yr_offset` that is intended to apply to all dates and observations.  This option should be removed and possibly be replaced by an input and output reference date for each model run and observational data set.  Discussion is needed if these reference dates are actually needed once we find a more generalized calendar (e.g. `netcdftime`).
```

Comment:
We need to figure out what to do about `yr_offset` and related things like that because this appears to be a potential problem that may be causing unnecessary roadblocks.